jina/flow/asyncio.py

```diff
@@ -0,0 +1,299 @@
+from typing import Union, List, Iterator, Callable
+from urllib.request import Request
+
+from . import Flow
+from ..clients import InputFnType
+from ..clients.asyncio import AsyncClient
+
+if False:
+    import numpy as np
+
+
+class AsyncFlow(Flow):
```

Comment:
Should we implement an async context manager for AsyncFlow? 
```
async with AsyncFlow().add() as f:
   await f.index(['abc'])
```

Reply:
no. `async with` is only implemented when one wants to waive the control during `__enter__` `__exit__`. We don't have such needs.

Reply:
This is because we have taken special care not to propagate the `async...await` to the caller. e.g. - `Client` and `AsyncClient` should be context managers, so that they can be used outside a Flow with ease. Similarly `GatewayPea`  & `RESTGatewayPea` can be an async context manager, even though I don't see heavy usage for the same.